doc/chapters/debugging/index.rst

```diff
@@ -0,0 +1,197 @@
+
+.. _debugging:
+
+Debugging
+*********
+
+The use of containers can make debugging a bit harder as we sometimes need to
+get information from different sources of information. Paths can require some
+help to parse for the computer as well as yourself
+
+Debugging applications running in containers
+============================================
+
+This section and subsections describes some common ways of debugging
+applications running inside containers and how to use common tools on the
+applications running inside the containers.
+
+LXC
+---
+
+LXC provides several tools that can be used to check out the containers
+currently running on the system. `Note that all commands must be run as the main
+user which the containers are started as, ie root usually.`
+
+*lxc-ls* lists all currently running containers on the host.
+
+*lxc-attach* can be used to "get inside" the container with a shell for example,
+or to test run commands on the inside of the container. Please note that the
+command given must be available inside the container.
+
+*lxc-console* can be used to login with a shell inside the container in question.
+
+*lxc-info* provides information about the running container, such as network
+interface used for t he bridge, the state of the container, what PID it has and
+so forth.
+
+Cleanup
+-------
+
+The internal filesystem will always be deleted and recreated when an app is
+restarted. This means that anything you want to save during debugging needs to
+be saved to a bind mounted directory before you shut down the container.
```

Comment:
Perhaps a bit misleading wording. Restarting an app is not necessarily the same as destroying the container and starting a new one, although that is probably what will happen most of the time.